src/patient-registration/input/custom-input/identifier/identifier-input.component.tsx

```diff
@@ -44,27 +44,32 @@ export const IdentifierInput: React.FC<IdentifierInputProps> = ({
     if (identifierType.format) {
       validatorProps = validatorProps.matches(new RegExp(identifierType.format), 'Invalid identifier format!');
     }
+    let schemaBuilder = {};
+    schemaBuilder[identifierType.fieldName] = validatorProps;
     identifierValidationSchema[identifierType.fieldName] = validatorProps;
-    setValidationSchema(validationSchema.concat(Yup.object(identifierValidationSchema)));
+    setIdentifierValidationSchema(Yup.object(schemaBuilder));
   }, []);
 
   useEffect(() => {
-    const selectedSource = find(sources, { name: selectSourceField.value });
-
-    if (selectedSource && selectedSource.autoGenerationOption) {
-      setAutoGenerationOption(selectedSource.autoGenerationOption);
-      if (selectedSource.autoGenerationOption.automaticGenerationEnabled) {
-        identifierType.autoGenerationSource = selectedSource;
-        if (validationSchema.fields[identifierType.fieldName]) {
-          validationSchema.fields[identifierType.fieldName] = Yup.string();
+    if (selectSourceField.value) {
+      const selectedSource = find(sources, { name: selectSourceField.value });
+      if (selectedSource && selectedSource.autoGenerationOption) {
+        setAutoGenerationOption(selectedSource.autoGenerationOption);
+        if (selectedSource.autoGenerationOption.automaticGenerationEnabled) {
+          identifierType.autoGenerationSource = selectedSource;
+          if (validationSchema.fields[identifierType.fieldName]) {
+            validationSchema.fields[identifierType.fieldName] = Yup.string();
+          }
+        } else {
+          setValidationSchema(validationSchema.concat(identifierValidationSchema));
```

Comment:
I kind of missed this in your earlier PR but while reviewing I felt this method was named wrongly. This should have been something like `onValidationSchemaChange` and this [code](https://github.com/openmrs/openmrs-esm-patient-registration/blob/6e182e3f7ff755b653ae524bc160357269d19321/src/patient-registration/patient-registration.component.tsx#L276-L280)

```javascript
<IdentifierSection
    identifierTypes={identifierTypes}
    validationSchema={validationSchema}
    setValidationSchema={setValidationSchema}
/>
```

Should have been 

```javascript
<IdentifierSection
    identifierTypes={identifierTypes}
    validationSchema={validationSchema}
    onValidationSchemaChange={newSchema => setValidationSchema(validationSchema.concat(schema))}
/>
```

Or define a `handleValidationSchema` function 

```javascript
const handleValidationSchemaChange = (newSchema) => {
    setValidationSchema(prevSchema => prevSchema.concat(newSchema))
}
```

and then use it in the component like so 

```javascript
<IdentifierSection
    identifierTypes={identifierTypes}
    validationSchema={validationSchema}
    setValidationSchema={handleValidationSchema}
/>
```

The above suggestions does not change the logic of your code. It just makes it more readable since it's following widely accepted patterns. When I first saw you call `setValidationSchema` here my first instinct was that this component `IndentifierInput` was managing a state called `validationSchema` until I looked at existing code not part of this PR to understand that it was just lifting the state.

This is just a note and since your code works and has already been approved you don't have to worry about this but in future, I think following this pattern will make your code more obvious.

Reply:
@ivange94 thanks for the suggestion, it looks great. I think I will merge this fix in for now, since it's like "first aid" to the broken parts in the server. I will surely consider to refactor the whole logic to the above patterns.